.github/workflows/build.yaml

```diff
@@ -0,0 +1,43 @@
+name: Build & Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  doit:
+    name: 'Build & Test: ${{matrix.cfg.name}}'
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - cfg: {name: 'macos',             runson: 'macos-latest'}
+          - cfg: {name: 'gcc10',             runson: 'ubuntu-latest', container: 'debian:bullseye'}
+          - cfg: {name: 'gcc11',             runson: 'ubuntu-latest', container: 'ubuntu:jammy'}
+          - cfg: {name: 'gcc12',             runson: 'ubuntu-latest', container: 'debian:bookworm'}
+          - cfg: {name: 'gcc-arch',          runson: 'ubuntu-latest', container: 'archlinux'}
```

Comment:
I wanted to test the latest gcc & clang in a simple manner, so that is why Arch is used here. That's not perfect -- Arch is still on clang15. But it also means CI isn't repeatable across a long time frame either, as compiler versions could shift over time. I'm willing to accept that trade off for being able to run on latest compilers easily.